src/seabreeze/cseabreeze/c_seabreeze_wrapper.pyx

```diff
@@ -841,14 +861,112 @@ cdef class SeaBreezeSpectrometerFeature(SeaBreezeFeature):
+                last_spectrum_count
+                last_microsecond_count
+                scans_to_average
+                intensities
+        """
+        cdef int error_code
+        cdef int bytes_written
+        cdef unsigned char* c_buffer = NULL
+        cdef int out_length
+        cdef int sample_number
+        sample_number = int(number_of_samples)
+
+        out_length = (self._spectrum_length*2 + 68) * number_of_samples
+
+        c_buffer = <unsigned char*> PyMem_Malloc(out_length * sizeof(unsigned char))
+        if not c_buffer:
+            raise MemoryError("could not allocate memory for buffer")
+        try:
+            bytes_written = self.sbapi.spectrometerGetFastBufferSpectrum(self.device_id, self.feature_id, &error_code,
+                                                                        &c_buffer[0], out_length, sample_number)
+            if error_code != 0:
+                raise SeaBreezeError(error_code=error_code)
+            data = c_buffer[:bytes_written]
+        finally:
+            PyMem_Free(c_buffer)
+
+        # Define Dataset structure for individual measurements.
+        Dataset = namedtuple("Dataset", [
+                                        "metadata_protocol_version",
+                                        "metadata_length",
+                                        "pixel_data_length",
+                                        "microsecond_counter",
+                                        "integration_time_micros",
+                                        "pixel_data_format",
+                                        "spectrum_count",
+                                        "last_spectrum_count",
+                                        "last_microsecond_count",
+                                        "scans_to_average",
+                                        "intensities"
+                                        ])
+        buffer_data = []
+        offset = 0
+        for i in range(sample_number):
+            # decode metadata
+            protocol_version = np.frombuffer(data[offset+0:offset+2], dtype=np.uint16)[0]
+            metadata_length = np.frombuffer(data[offset+2:offset+4], dtype=np.uint16)[0]
+            pixel_data_length = np.frombuffer(data[offset+4:offset+8], dtype=np.uint32)[0]
+            microsecond_counter = np.frombuffer(data[offset+8:offset+16], dtype=np.uint64)[0]
+            integration_time_micros = np.frombuffer(data[offset+16:offset+20], dtype=np.uint32)[0]
+            pixel_data_format = np.frombuffer(data[offset+20:offset+24], dtype=np.uint32)[0]
+            spectrum_count = np.frombuffer(data[offset+24:offset+28], dtype=np.uint32)[0]
+            last_spectrum_count = np.frombuffer(data[offset+28:offset+32], dtype=np.uint32)[0]
+            last_microsecond_count = np.frombuffer(data[offset+32:offset+40], dtype=np.uint64)[0]
+            scans_to_average = np.frombuffer(data[offset+40:offset+42], dtype=np.uint16)[0]
```

Comment:
to decode the metadata it would be better to use struct.unpack and then to feed the resulting tuple into the NamedTuple that encapsulates the metadata.

Please move the namedtuple definition outside of the function. It should be globally available in the module to prevent that every function call creates the class.